### PR TITLE
Horizontal bar chart implimentation

### PR DESCRIPTION
**Pull request by deepaknevdepsl**
### Description
Added Horizontal Bar chart as a new visualization  type.

### Issues Resolved
#885

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx [hunk 1]
```diff
@@ -221,8 +222,13 @@ export const ConfigPanel = ({ visualizations, setCurVisId, callback, changeIsVal
   };
 
   const memorizedVisualizationTypes = useMemo(() => {
```
Comment (abasatwar) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx:224]: can we avoid usememo here?

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx [hunk 2]
```diff
@@ -221,8 +222,13 @@ export const ConfigPanel = ({ visualizations, setCurVisId, callback, changeIsVal
...
+    let visDefinition = {}
     return ENABLED_VIS_TYPES.map((vs: string) => {
-      const visDefinition = getVisType(vs);
+      if (vs === visChartTypes.Bar || vs === visChartTypes.HorizontalBar) {
+        visDefinition = vs === visChartTypes.Bar ? getVisType(vs, { type: visChartTypes.Bar }) : getVisType(vs, { type: visChartTypes.HorizontalBar });
```
Comment (abasatwar) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx:228]: check the formatting of all files and lint errors too if any.

**Review by abasatwar — CHANGES_REQUESTED** [review]

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx (hunk 1 above)
Comment (deepaknevdepsl) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx:224]: No, as we are using memorized data here.

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx (hunk 2 above)
Comment (deepaknevdepsl) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx:228]: formatted files and pushed.

**Comment by abasatwar** [comment]
https://github.com/opensearch-project/observability/pull/924 - is raised for this functionality

**Pull request closed without merging**